### PR TITLE
Improved Gunnery Algorithm for Projectiles with Drag

### PR DESCRIPTION
**Pull request by NCGThompson**
In response to issue #3686.
This causes turrets to correctly compensate for turret's velocity. It does so by finding the amount of drift caused by the velocity for the calculated rendezvous time. Since the correct rendezvous time is also dependent on offset of the turret velocity, they are calculated recursively with a small number of repetitions.
The math is explained [in this video](https://youtu.be/pOFl0BhUkDI?t=6m49s).
Balance compensation for the Korath Grab-Strike may be worth consideration.
The amount of repetitions can be changed by changing the initial value of 'iterations'.

### CODE REVIEW

File: source/AI.cpp [hunk 1]
```diff
@@ -2442,39 +2442,60 @@ void AI::AimTurrets(const Ship &ship, Command &command, bool opportunistic) cons
...
-				// weight. Outside that range, give them lower priority.
-				rendezvousTime = max(0., rendezvousTime - weapon->TotalLifetime());
-				// Always prefer ships that you are able to hit.
-				double score = turnTime + (180. / weapon->TurretTurn()) * rendezvousTime;
-				if(score < bestScore)
-				{
-					bestScore = score;
-					bestAngle = degrees;
-				}
+                Point p = target->Position() - start;
```
Comment (tehhowch) [inline source/AI.cpp]: This is a `tabs` repo :)

For comments, they should be sentences and start with a space after the `//`

File: source/AI.cpp [hunk 2]
```diff
@@ -2456,7 +2456,7 @@ void AI::AimTurrets(const Ship &ship, Command &command, bool opportunistic) cons
                 double rendezvousTime = 0.;
                 if(weapon->Acceleration())
                 {
-                    //Uses iterative approxamation for projectiles with drag.
+                    // For projectiles with drag, use iterative approxamation.
```
Comment (Bladewood) [inline source/AI.cpp]: approximation

File: source/AI.cpp (hunk 1 above)
Comment (Bladewood) [reply source/AI.cpp]: Please use actual tabs, not space quadruplets or another form of indentation.

**Comment by Amazinite** [comment]
Closing as this has been marked unlikely.

**Pull request closed without merging**